[Win] Initialize RunLoop class in WTF::initialize instead of WTF::initializeMainThread

fujii opened the pull request:
#### 536794017a946bb43409d38bc975b04baedaf8e0
<pre>
[Win] Initialize RunLoop class in WTF::initialize instead of WTF::initializeMainThread
<a href="https://bugs.webkit.org/show_bug.cgi?id=255903">https://bugs.webkit.org/show_bug.cgi?id=255903</a>

Reviewed by Ross Kirsling.

testapi.exe and testRegExp.exe didn&apos;t work after 263194@main because
RunLoop class wasn&apos;t initialized. They don&apos;t call
WTF::initializeMainThread. They call WTF::initialize instead.

Moved the callsite of RunLoop::registerRunLoopMessageWindowClass from
WTF::initializeMainThread to WTF::initialize.

* Source/WTF/wtf/Threading.cpp:
(WTF::initialize):
* Source/WTF/wtf/win/MainThreadWin.cpp:
(WTF::initializeMainThreadPlatform):

Canonical link: <a href="https://commits.webkit.org/263352@main">https://commits.webkit.org/263352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e5cbdaf78e7fa83d710c2961388a89491ba84e3f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4336 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4453 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4580 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5810 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4554 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4568 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4419 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/4779 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4396 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/4545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/3902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5809 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/2041 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/3879 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/7171 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/3600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/3871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/3946 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/5475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4120 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4353 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3536 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4434 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3872 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/3877 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/1103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/7928 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4535 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/499 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4226 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/3/builds/1210 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WTF/wtf/Threading.cpp:
```diff
@@ -57,6 +57,10 @@
 
+#if OS(WINDOWS)
+#include <wtf/RunLoop.h>
+#endif
```
I don't think you need an `#if` here; the header isn't platform-specific, even if the usage happens to be.

fujii replied:
Fixed.

rkirsling reviewed (approved):
r=me with a comment

fujii committed:
[Win] Initialize RunLoop class in WTF::initialize instead of WTF::initializeMainThread
https://bugs.webkit.org/show_bug.cgi?id=255903

Reviewed by Ross Kirsling.

testapi.exe and testRegExp.exe didn't work after 263194@main because
RunLoop class wasn't initialized. They don't call
WTF::initializeMainThread. They call WTF::initialize instead.

Moved the callsite of RunLoop::registerRunLoopMessageWindowClass from
WTF::initializeMainThread to WTF::initialize.

* Source/WTF/wtf/Threading.cpp:
(WTF::initialize):
* Source/WTF/wtf/win/MainThreadWin.cpp:
(WTF::initializeMainThreadPlatform):

Canonical link: https://commits.webkit.org/263352@main
